packages/jupyter-ai/src/completions/plugin.ts

```diff
@@ -68,28 +88,60 @@ export const inlineCompletionProvider: JupyterFrontEndPlugin<void> = {
       return languageRegistry.findByMIME(editor.model.mimeType);
     };
 
-    let settings: ISettingRegistry.ISettings | null = null;
+    // ic := inline completion
+    let icSettings: IcPluginSettings | null = null;
+
+    // jaiIcp := Jupyter AI inline completion provider
+    // if not defined, the default settings are used
+    let jaiIcpSettings = JaiInlineProvider.DEFAULT_SETTINGS;
 
+    // Reflect changes made by user from Settings Editor
     settingRegistry.pluginChanged.connect(async (_emitter, plugin) => {
-      if (plugin === INLINE_COMPLETER_PLUGIN) {
-        // Only load the settings once the plugin settings were transformed
-        settings = await settingRegistry.load(INLINE_COMPLETER_PLUGIN);
+      if (plugin !== INLINE_COMPLETER_PLUGIN) {
+        return;
       }
+      icSettings = (await settingRegistry.load(
+        INLINE_COMPLETER_PLUGIN
+      )) as IcPluginSettings;
+      jaiIcpSettings = icSettings.composite.providers[JaiInlineProvider.ID];
+      console.log(jaiIcpSettings);
```

Comment:
```suggestion
```

but you may want to have a look at the console to see how this value changes as (a) lab is loaded (b) user toggles options from statusbar menu (c) user makes edits via settings editor